### ISSUE TITLE
Aspect processing issues

### ISSUE THREAD

**Issue by samaloney** [i4Ds/STIXCore#209]
> I have recently mirrored the L1 FITS files from pub099/…/SCI/ to our server at AIP, and I find many issues with the aspect burst data files. It would take me some time to make a detailed list with all issues, here’s a general description of some problems:
>
> - many files are missing, although the “L1A” version exists on pub023
> - the time resolution is often wrong, but not always. For example, quite a few files appear to have a time resolution of 3.0s or 
> even 10.0s but actually the maximum value that I’ve used is 1.024 s
> - when several requests were submitted for different time ranges in a given day, the data is merged in a file and the time 
> information is not correct
>
> Here is a list of all requests that were submitted since day 1, with all parameters and whether they failed or were properly > executed:
> https://docs.google.com/spreadsheets/d/1SBtpKrkXP6t3NfQA6Bl3nEpglR5W85R3gLmqRtikLoQ/edit?usp=sharing

**Comment by samaloney** [comment]
@FredSchuller could you expand a little on the the google sheets for example the number of data samples you expect, the time resolution in seconds?

Also it's not clear to me which rows should have fits files and which should not if you could indicate that would be really helpful. I see you have column with unique ID but there are entries way before aspect data had unique request IDs so I'm a little confused?

**Comment by samaloney** [comment]
This is the list of aspect files produced by the current pipeline

> ./2020/04/15/SCI/solo_L1_stix-sci-aspect-burst_20200415T135524-20200415T135532_V01_0.fits
./2020/04/15/SCI/solo_L1_stix-sci-aspect-burst_20200415T175903-20200415T175911_V01_0.fits
./2020/04/17/SCI/solo_L1_stix-sci-aspect-burst_20200417T215708-20200417T215716_V01_0.fits
./2020/04/27/SCI/solo_L1_stix-sci-aspect-burst_20200427T215140-20200427T215148_V01_0.fits
./2020/04/29/SCI/solo_L1_stix-sci-aspect-burst_20200429T214305-20200429T214313_V01_0.fits
./2020/05/06/SCI/solo_L1_stix-sci-aspect-burst_20200506T190738-20200506T190811_V01_0.fits
./2020/05/08/SCI/solo_L1_stix-sci-aspect-burst_20200508T082331-20200508T082339_V01_0.fits
./2020/05/16/SCI/solo_L1_stix-sci-aspect-burst_20200516T202523-20200516T203406_V01_0.fits
./2020/05/27/SCI/solo_L1_stix-sci-aspect-burst_20200527T084325-20200527T084535_V01_0.fits
./2020/05/28/SCI/solo_L1_stix-sci-aspect-burst_20200528T154946-20200528T154954_V01_0.fits
./2020/05/28/SCI/solo_L1_stix-sci-aspect-burst_20200528T174201-20200528T174209_V01_0.fits
./2020/06/14/SCI/solo_L1_stix-sci-aspect-burst_20200614T172736-20200614T173618_V01_0.fits
./2020/06/20/SCI/solo_L1_stix-sci-aspect-burst_20200620T014130-20200620T014341_V01_0.fits
./2021/01/07/SCI/solo_L1_stix-sci-aspect-burst_20210107T155334-20210107T160217_V01_0.fits
./2021/02/20/SCI/solo_L1_stix-sci-aspect-burst_20210220T055123-20210220T060005_V01_0.fits
./2021/02/21/SCI/solo_L1_stix-sci-aspect-burst_20210221T054449-20210221T054700_V01_0.fits
./2021/02/21/SCI/solo_L1_stix-sci-aspect-burst_20210221T055000-20210221T055210_V01_0.fits
./2021/03/23/SCI/solo_L1_stix-sci-aspect-burst_20210323T032808-20210323T033651_V01_0.fits
./2021/03/26/SCI/solo_L1_stix-sci-aspect-burst_20210326T182659-20210326T182732_V01_0.fits
./2021/07/21/SCI/solo_L1_stix-sci-aspect-burst_20210720T235959-20210721T000841_V01_0.fits
./2021/08/02/SCI/solo_L1_stix-sci-aspect-burst_20210802T065359-20210802T065407_V01_0.fits
./2021/08/09/SCI/solo_L1_stix-sci-aspect-burst_20210809T155958-20210809T160841_V01_0.fits
./2021/08/17/SCI/solo_L1_stix-sci-aspect-burst_20210817T024159-20210817T024304_V01_0.fits
./2021/08/17/SCI/solo_L1_stix-sci-aspect-burst_20210817T024359-20210817T025241_V01_2108170028.fits
./2021/08/17/SCI/solo_L1_stix-sci-aspect-burst_20210817T042859-20210817T043741_V01_2108170029.fits
./2021/08/17/SCI/solo_L1_stix-sci-aspect-burst_20210817T061359-20210817T062241_V01_2108170030.fits
./2021/08/17/SCI/solo_L1_stix-sci-aspect-burst_20210817T075900-20210817T080742_V01_2108170031.fits
./2021/08/17/SCI/solo_L1_stix-sci-aspect-burst_20210817T094359-20210817T095242_V01_2108170032.fits
./2021/09/13/SCI/solo_L1_stix-sci-aspect-burst_20210913T055800-20210913T060010_V01_0.fits
./2021/09/14/SCI/solo_L1_stix-sci-aspect-burst_20210914T024359-20210914T024449_V01_2109140643.fits
./2021/09/14/SCI/solo_L1_stix-sci-aspect-burst_20210914T042859-20210914T043741_V01_2109140644.fits
./2021/09/14/SCI/solo_L1_stix-sci-aspect-burst_20210914T061359-20210914T062241_V01_2109140645.fits
./2021/09/14/SCI/solo_L1_stix-sci-aspect-burst_20210914T075859-20210914T080741_V01_2109140646.fits
./2021/09/14/SCI/solo_L1_stix-sci-aspect-burst_20210914T094359-20210914T095241_V01_2109140647.fits
./2021/10/13/SCI/solo_L1_stix-sci-aspect-burst_20211013T034959-20211013T035842_V01_0.fits
./2021/10/13/SCI/solo_L1_stix-sci-aspect-burst_20211013T034959-20211013T035842_V01_2110130059.fits
./2021/10/13/SCI/solo_L1_stix-sci-aspect-burst_20211013T035000-20211013T035121_V01_2110130054.fits
./2021/10/13/SCI/solo_L1_stix-sci-aspect-burst_20211013T154959-20211013T155121_V01_2110130055.fits
./2021/10/13/SCI/solo_L1_stix-sci-aspect-burst_20211013T154959-20211013T155841_V01_2110130060.fits
./2021/11/30/SCI/solo_L1_stix-sci-aspect-burst_20211129T235959-20211130T000120_V01_2111300007.fits
./2021/11/30/SCI/solo_L1_stix-sci-aspect-burst_20211130T000000-20211130T000210_V01_0.fits
./2021/12/09/SCI/solo_L1_stix-sci-aspect-burst_20211209T165759-20211209T165829_V01_2112090010.fits

**Comment by FredSchuller** [comment]
@samaloney The time resolution (in ms) is given in column H ("sampling") in the google sheet. I have now added a column "Nb. rows (exp'd)" that gives the expected number of entries (= duration / sampling).

When the execution date is marked in green, this means that I was able to download the L1A data from pub023; same is true when it's marked in yellow (usually this means there are data, but incomplete). Only when it's red there was a good reason not to get data (e.g. if STIX was turned off (!)). For example in April 2021 there were several requests which returned data, but there is no file on pub099. I can go through your list to compile the list of missing ones.

Regarding the "unique ID", this is from column "UID" in the "STIX user data requests" table
https://pub023.cs.technik.fhnw.ch/view/user-data-requests/list
but since this ID was not really used in the past, at some point I stopped reporting this value in my google table. Only since the latest FSW update would it make sense to keep track of this ID.
